### PR TITLE
[tx] add optimizations for selective log softmax

### PR DESCRIPTION
**Pull request by aditchawdhary**
Summary
Optimizes the selective log-softmax operation in the training loss computation, achieving a 41% throughput improvement on Qwen3-0.6B.

Changes
- Added `selective_log_softmax_jax` kernel that computes log probabilities only for target tokens
- Avoids materializing the full `[B, T, V]` log_softmax array by computing `log_softmax(logits)[target_ids]` directly
- Integrated into `engine.py` loss computation path

## Performance
Benchmarked on L40S GPU with Qwen3-0.6B (151K vocab):
- **Before:** ~5,647 tokens/sec
- **After:** 7,984 tokens/sec
- **Improvement:** +41% throughput, -36% time per step

This eliminates the memory and compute overhead of the full vocabulary softmax when only target token probabilities are needed.

================================================================================

RESULTS

================================================================================

steps:       20

elapsed:     5.130 s

steps/sec:   3.90

tokens/sec:  7984

ms/step:     256.51

================================================================================

### CODE REVIEW

File: skyrl-tx/tx/kernels/selective_logsoftmax.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        [B, T] log probabilities at target indices
+    """
+    B, T, V = logits.shape
+    logits_flat = logits.reshape(-1, V)
+    target_ids_flat = target_ids.reshape(-1)
+    
+    # Compute log_sum_exp for numerical stability
+    max_logits = jnp.max(logits_flat, axis=-1, keepdims=True)
+    exp_sum = jnp.sum(jnp.exp(logits_flat - max_logits), axis=-1, keepdims=True)
+    log_sum_exp = max_logits + jnp.log(exp_sum)
```
[the remark this answers is not in the document]
Comment (aditchawdhary) [reply skyrl-tx/tx/kernels/selective_logsoftmax.py]: thanks gemini bot, I have used jax.scipy.special.logsumexp now! ❤️

File: skyrl-tx/tx/kernels/selective_logsoftmax.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+"""Selective log-softmax kernel for JAX."""
+
+import jax
+import jax.numpy as jnp
+from functools import partial
+
+
+@partial(jax.jit, static_argnames=['chunk_size'])
```
[the remark this answers is not in the document]
Comment (aditchawdhary) [reply skyrl-tx/tx/kernels/selective_logsoftmax.py]: thanks gemini bot, fixed this

**Comment by aditchawdhary** [comment]
@pcmoritz  can you take a look and review the changes?
